lib/rule-helpers.js

```diff
@@ -243,10 +246,17 @@ module.exports.noPropertyAccessDuringSSR = function noPropertyAccessDuringSSR(
                 return;
             }
             if (
+                node.parent.type === 'CallExpression' &&
+                node.parent.optional !== true &&
                 node.object.type === 'ThisExpression' &&
                 node.property.type === 'Identifier' &&
                 forbiddenPropertyNames.includes(node.property.name)
             ) {
+                // Prevents expressions like:
+                // this.addEventListener('click', () => { ... });
```

Comment:
Is this a typo, and should be `dispatchEvent` or something? This should function fine in SSR, where `addEventListener` is a no-op.

Reply:
It's actually a leftover of a former "incarnation" of the PR. In my first version I added `addEventListener` to the disallowed prop as I thought it was just a miss that it's not there. That was before you pointed me to the runtime code with all the "polyfills" on that end. There I saw that actually `removeEventListener` could be removed from that list. I'll update the comment.